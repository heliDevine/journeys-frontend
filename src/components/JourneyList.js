import React, { useState } from 'react';
import moment from 'moment';
import JourneyPopup from './JourneyPopup';

const JourneyList = ({ journeys }) => {
  const [selectedJourney, setSelectedJourney] = useState(null);
  const [showPopup, setShowPopup] = useState(false);

  const handleJourneySelect = journey => {
    setSelectedJourney(journey);
    setShowPopup(true);
  };

  const handleClosePopup = () => {
    setSelectedJourney(null);
    setShowPopup(false);
  };

  return (
    <div className="journey_grid_container">
      {journeys.map(journey => (
        <div className="journey-summary" key={journey.id}>
          <div className="journey-text">
            Journey on {moment(journey.departureTime).format('ddd Do MMM YYYY')}{' '}
            at {moment(journey.departureTime).format('HH:mm')} from{' '}
            {journey.departureStationName} to {journey.returnStationName} was{' '}
            <button
              className="info-button"
              type="button"
              onClick={() => handleJourneySelect(journey)}
            >
              More Info
            </button>
          </div>
        </div>
      ))}
      {showPopup && selectedJourney && (
        <JourneyPopup
          selectedJourney={selectedJourney}
          onClose={handleClosePopup}
        />
      )}
    </div>
  );
};

export default JourneyList;
