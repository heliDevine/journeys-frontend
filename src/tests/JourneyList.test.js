import React from 'react';
import { render } from '@testing-library/react';
import JourneyList from '.././components/JourneyList';

describe('JourneyList', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<JourneyList />);

    expect(asFragment(<JourneyList />)).toMatchSnapshot();
  });
});
