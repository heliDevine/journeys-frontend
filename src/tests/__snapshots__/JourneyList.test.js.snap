// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JourneyList renders correctly 1`] = `
<DocumentFragment>
  <div
    class="journey_grid_container"
  >
    <div
      class="journey-summary"
    >
      <div
        class="journey-text"
      >
        Invalid date at Invalid date from testName to TestName2 
        <button
          class="info-button"
          type="button"
        >
          more details
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
